core/ledger/pvtdatastorage/snapshot_data_importer_test.go

```diff
@@ -0,0 +1,1034 @@
+	testDir := testDir(t)
+	defer os.RemoveAll(testDir)
+
+	configHistoryMgr, err := confighistorytest.NewMgr(testDir)
+	require.NoError(t, err)
+	defer configHistoryMgr.Close()
+
+	// setup a sample config history for namespace1
+	err = configHistoryMgr.Setup("test-ledger", "namespace1",
+		map[uint64][]*peer.StaticCollectionConfig{
+			15: {
+				{
+					Name:             "coll1",
+					MemberOrgsPolicy: iamIn.toMemberOrgPolicy(),
+					BlockToLive:      0,
+				},
+				{
+					Name:             "coll2",
+					MemberOrgsPolicy: iamOut.toMemberOrgPolicy(),
+					BlockToLive:      100,
+				},
+				{
+					Name:             "coll3",
+					MemberOrgsPolicy: iamIn.toMemberOrgPolicy(),
+					BlockToLive:      500,
+				},
+			},
+
+			10: {
+				{
+					Name:             "coll1",
+					MemberOrgsPolicy: iamOut.toMemberOrgPolicy(),
+				},
+				{
+					Name:             "coll2",
+					MemberOrgsPolicy: iamIn.toMemberOrgPolicy(),
+				},
+			},
+
+			5: {
+				{
+					Name:             "coll1",
+					MemberOrgsPolicy: iamIn.toMemberOrgPolicy(),
+				},
+			},
+		},
+	)
+	require.NoError(t, err)
+
+	// setup a sample config history for namespace2
+	err = configHistoryMgr.Setup("test-ledger", "namespace2",
+		map[uint64][]*peer.StaticCollectionConfig{
+			50: {
+				{
+					Name:             "coll1",
+					MemberOrgsPolicy: iamIn.toMemberOrgPolicy(),
+					BlockToLive:      300,
+				},
+			},
+		},
+	)
+	require.NoError(t, err)
+
+	eligibilityBTLCache := eligibilityBTLCache{
+		membershipProvider:     newMockMembershipProvider("myOrg"),
+		configHistoryRetriever: configHistoryMgr.GetRetriever("test-ledger"),
+		eligibilityHistory:     map[nsColl][]*eligibility{},
+		btl:                    map[nsColl]uint64{},
+	}
+
+	err = eligibilityBTLCache.loadDataFor("namespace1")
+	require.NoError(t, err)
+	err = eligibilityBTLCache.loadDataFor("namespace2")
+	require.NoError(t, err)
+
+	require.Equal(
+		t,
+		map[nsColl]uint64{
+			{ns: "namespace1", coll: "coll2"}: 100,
+			{"namespace1", "coll3"}:           500,
+			{"namespace2", "coll1"}:           300,
+		},
+		eligibilityBTLCache.btl,
+	)
+
+	require.Equal(
+		t,
+		map[nsColl][]*eligibility{
+			{ns: "namespace1", coll: "coll1"}: {{configBlockNum: 15, isEligible: true}, {10, false}, {5, true}},
+			{"namespace1", "coll2"}:           {{15, false}, {10, true}},
+			{"namespace1", "coll3"}:           {{15, true}},
+			{"namespace2", "coll1"}:           {{50, true}},
+		},
+		eligibilityBTLCache.eligibilityHistory,
+	)
+}
+
+func TestEligibilityBTLCacheEligibleExplicitCollection(t *testing.T) {
+	eligibilityBTLCache := eligibilityBTLCache{
+		membershipProvider: newMockMembershipProvider("myOrg"),
+	}
+
+	t.Run("ineligible-then-eligible", func(t *testing.T) {
+		eligibilityBTLCache.eligibilityHistory = map[nsColl][]*eligibility{
+			{ns: "ns", coll: "coll"}: {
+				{configBlockNum: 10, isEligible: true},
+				{configBlockNum: 5, isEligible: false},
+			},
+		}
+
+		for i := 6; i < 15; i++ {
```

Comment:
even for 1 to 5, this test would pass, right?

Reply:
No, it should expect an error for that range - `unexpected error - no collection config found below block number ...`